File: 序号/2611.老鼠和奶酪.py
class Solution:
    def miceAndCheese(self, reward1: List[int], reward2: List[int], k: int) -> int:
        n = len(reward1)
        # 1号能在n中选k个，要让总值最大，判断出哪些奶酪让1号吃更好，计算出差值
        l=[]
        for i in range(n):
            diff = reward1[i]-reward2[i]
            l.append([diff,reward1[i],reward2[i]])
        l.sort(reverse=True)
        ans=0
        for i in range(k):
            ans+=l[i][1]
        for i in range(k,n):
            ans+=l[i][2]
        return ans

File: 序号/2611. 老鼠和奶酪.cpp
class Solution {
 public:
  int miceAndCheese(vector<int>& reward1, vector<int>& reward2, int k) {
    // 吃k块奶酪最大得分
    vector<vector<int>> v;
    int n=reward1.size();
    for(int i=0;i<n;i++){
      int a=reward1[i],b=reward2[i];
      vector<int> cur{a-b,a,b};
      v.emplace_back(cur);     
    }
    int ans=0;
    sort(v.begin(),v.end(),[](vector<int> const &a,vector<int> const &b){return b[0]<a[0];});
    for(int i=0;i<n;i++){
      if(i<k){
        ans+=v[i][1];
      }else{
        ans+=v[i][2];
      }
    }
    return ans;
}